protocols/kad/src/behaviour.rs

```diff
@@ -2674,6 +2683,12 @@ pub enum KademliaEvent {
     /// See [`Kademlia::kbucket`] for insight into the contents of
     /// the k-bucket of `peer`.
     PendingRoutablePeer { peer: PeerId, address: Multiaddr },
+
+    /// This peer's mode has been updated.
+    ///
+    /// This usually happens in response to an address change or a new external
+    /// address being added or removed.
```

Comment:
```suggestion
    /// This happens in response to an external
    /// address being added or removed.
```

This is the only reason right? If so, let's be specific in the wording here.